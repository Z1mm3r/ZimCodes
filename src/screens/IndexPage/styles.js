const styles = {
    gridContainer:{
        display: 'flex',
    },
    container:{
        width:"100%",
        height:"100%",
    },
    paraWrap:{
        width:"100vw",
        height:"85vh"
    },
    miniText:{
        fontSize:"4em"
    }
}

export default styles