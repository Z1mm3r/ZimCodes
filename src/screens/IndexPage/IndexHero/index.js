import React from 'react'
import {makeStyles} from '@material-ui/core'

import IndexHero from './IndexHero.js'
import styles from './styles.js'


const useStyle = makeStyles(styles)

const Index =  function(props){
    const {
        lerpVal
    } = {...props}
    const classes = useStyle(props);
    return(
        <IndexHero {...props} classes={classes}/>
    )
}

export default Index