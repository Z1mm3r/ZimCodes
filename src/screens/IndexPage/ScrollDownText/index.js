import React from 'react'
import {makeStyles} from '@material-ui/core'

import ScrollDownText from './ScrollDownText'
import styles from './styles'

const useStyle = makeStyles(styles)

const Index = (props) =>{
    
    const{
        lerpVal
    } = {...props}
    
    const classes = useStyle();

    return(
        <ScrollDownText classes={classes} {...props}/>
    )
}

export default Index