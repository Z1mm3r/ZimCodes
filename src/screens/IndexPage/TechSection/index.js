import React from 'react'
import {makeStyles} from '@material-ui/core'

import TechSection from './TechSection.js'
import styles from './styles.js'


const useStyle = makeStyles(styles)

const Index =  function(props){
    const classes = useStyle(props);
    return(
        <TechSection classes={classes}/>
    )
}

export default Index