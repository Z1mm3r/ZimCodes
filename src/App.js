import {createMuiTheme, MuiThemeProvider} from '@material-ui/core/styles'
import CssBaseline from '@material-ui/core/CssBaseline'
import React, {useState, useEffect} from 'react'
import { Route, Switch, useLocation } from 'react-router-dom'
import {useTransition, animated} from '@react-spring/web'

import './App.css';
import AboutPage from './screens/AboutMePage'
import AboutSitePage from './screens/AboutSitePage'
import {DARK, LIGHT} from 'constants/colors'
import {darkTheme, lightTheme} from './themes'
import Header from './components/Header'
import IndexPage from './screens/IndexPage'
import LightingContext from './components/LightingContext'
import  {LOCATIONS } from 'constants/constants.js'
import {PAGE_LEFT_ENTER, PAGE_RIGHT_ENTER} from 'constants/transitions'
import ResumePage from './screens/ResumePage'

function App() {

  const [lighting,setLighting] = useState(LIGHT)

  //TODO debate pros and cons of this 
  //If we want React.strict mode... need to createMuiTheme everytime we want to swap from
  //light to dark.
  //IF we REMOVE React.strict... we can simply pass our themes in from the themes file
  //without using a themeState.
  //
  const [theme, setTheme] = useState(createMuiTheme(lightTheme))
  const toggleLighting = () => {
    setTheme( createMuiTheme (lighting === LIGHT ? darkTheme: lightTheme) )
    setLighting( lighting === LIGHT ? DARK : LIGHT)
  }
//----------------------Page Transition Logic--------------------------------------------------------------
  const location = useLocation();
  const [previousLocation, setPreviousLocation] = useState(location.pathname)
  const [entranceSide, setEntranceSide] = useState(PAGE_LEFT_ENTER)
  const [left,setLeft] = useState(true)
  //Using superSet to stop race conditions on our transition creations
  const [superSet, setSuperSet] = useState({
    location:location,
    previousLocation:previousLocation,
    left: left,
    entranceSide: entranceSide
   })

  useEffect(() => {
    if(LOCATIONS.indexOf(previousLocation) > LOCATIONS.indexOf(location.pathname)){
      setEntranceSide(PAGE_LEFT_ENTER)
      setLeft(true)
      setPreviousLocation(location.pathname)
    }
    else if(LOCATIONS.indexOf(previousLocation) < LOCATIONS.indexOf(location.pathname)){
      setEntranceSide(PAGE_RIGHT_ENTER)
      setLeft(false)
      setPreviousLocation(location.pathname)
    }
  }, [location])

  useEffect(() =>{
    setSuperSet({location,previousLocation,left,entranceSide})
  },[previousLocation])

  const transitions = useTransition(superSet.location,superSet.entranceSide,[superSet.location,superSet.entranceSide])[0]
//---------------------------------------------------------------------------------------

  return (
    <div className="App">
      <LightingContext.Provider value={{lighting,toggleLighting}}>
        <MuiThemeProvider theme={theme}>
            <CssBaseline />
            <Header/>
            {transitions((props,item) =>(
              <animated.div style={ props }>
                <Switch location={item}>
                  <Route exact path="/" component={IndexPage} />
                  <Route exact path="/about" component={AboutPage} />
                  <Route exact path="/resume" component={ResumePage} />
                  <Route exact path="/site" component={AboutSitePage} />
                </Switch>
              </animated.div>
            ))}
        </MuiThemeProvider  >
      </LightingContext.Provider>
    </div>
  );
}

export default App;
