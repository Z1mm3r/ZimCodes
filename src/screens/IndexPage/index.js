import React from 'react'
import Grid from '@material-ui/core/Grid'

import IndexHero from './IndexHero'
import AbsoluteWrapper from 'components/AbsoluteWrapper'




const IndexPage = (props) => {
    const{
        classes
    } = {...props}

    return(
        <AbsoluteWrapper>
            <Grid  container>
                <Grid item xs={12}>
                    Index Page
                </Grid>
                <Grid item xs={12}>
                    <IndexHero/>
                </Grid>
            </Grid>
        </AbsoluteWrapper>
    )
}

export default IndexPage


