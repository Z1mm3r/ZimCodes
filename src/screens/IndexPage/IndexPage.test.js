import IndexPage from './index'
import App from 'App.js'
import React from 'react'
import {screen} from '@testing-library/react'

import {renderWithRouter} from 'utils/testUtils.js'
  
  
  test('renders without crashing', () => {
    const {asFragment} = renderWithRouter(<App/>)
    expect(asFragment()).toMatchSnapshot();
  })
