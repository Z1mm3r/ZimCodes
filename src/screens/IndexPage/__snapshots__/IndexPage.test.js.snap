// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="makeStyles-lightMode-2"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
        />
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <div
            class="MuiGrid-root MuiGrid-container"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  Home
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
            >
              <a
                data-testid="navButtonLink"
                href="/about"
              >
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiButton-label"
                  >
                    About
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </a>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1"
            />
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
            >
              <div
                class="border"
              >
                <div>
                  <div>
                    <img
                      class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img"
                      src="/Images/WIP-LOGO.png"
                      title="Zim Codes"
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1"
            />
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
            >
              <a
                data-testid="navButtonLink"
                href="/resume"
              >
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiButton-label"
                  >
                    Resume
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </a>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
            >
              <a
                data-testid="navButtonLink"
                href="/site"
              >
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiButton-label"
                  >
                    About Site
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </a>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
        >
          <button
            class="MuiButtonBase-root MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root makeStyles-lightModeLightbulb-3 MuiSvgIcon-fontSizeLarge"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3.55 18.54l1.41 1.41 1.79-1.8-1.41-1.41-1.79 1.8zM11 22.45h2V19.5h-2v2.95zM4 10.5H1v2h3v-2zm11-4.19V1.5H9v4.81C7.21 7.35 6 9.28 6 11.5c0 3.31 2.69 6 6 6s6-2.69 6-6c0-2.22-1.21-4.15-3-5.19zm5 4.19v2h3v-2h-3zm-2.76 7.66l1.79 1.8 1.41-1.41-1.8-1.79-1.4 1.4z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      style="opacity: 1; transform: translate(0%,0);"
    >
      <div
        class="makeStyles-wrapper-13"
      >
        <div
          class="MuiGrid-root MuiGrid-container"
        >
          <div
            class="makeStyles-paraWrap-11"
          >
            <div
              class="test-name"
              style="position: absolute; width: 100%; height: 100%; overflow: scroll; overflow-y: scroll; overflow-x: hidden; transform: translate3d(0px,0px,0px);"
            >
              <div
                style="overflow: hidden; position: absolute; width: 100%; height: 0px; transform: translate3d(0px,0px,0px);"
              >
                <div
                  style="position: absolute; top: 0px; bottom: 0px; left: 0px; right: 0px; background-size: auto; background-repeat: no-repeat; will-change: transform; width: 100%; height: 0px; transform: translate3d(0,0px,0);"
                >
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                  >
                    <div
                      class="MuiGrid-root MuiGrid-container"
                    >
                      <div
                        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                      >
                        <div
                          class="MuiGrid-root MuiGrid-container MuiGrid-justify-content-xs-space-around"
                        >
                          <div
                            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
                          >
                            <div
                              class="MuiPaper-root MuiCard-root makeStyles-fullHeight-18 MuiPaper-elevation1 MuiPaper-rounded"
                            >
                              <div
                                align="center"
                                class="MuiGrid-root makeStyles-fullHeight-18 MuiGrid-container MuiGrid-direction-xs-column MuiGrid-justify-content-xs-center"
                              >
                                <div
                                  class="MuiGrid-root MuiGrid-item"
                                >
                                  <div
                                    class="MuiCardContent-root"
                                  >
                                    <div>
                                      <img
                                        class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img"
                                        src="/Images/Third-Party-Logos/ReactLogo.png"
                                        title="Zim Codes"
                                      />
                                    </div>
                                    <nav
                                      aria-label="main mailbox folders"
                                      class="MuiList-root MuiList-padding"
                                    >
                                      <h5
                                        class="MuiTypography-root MuiTypography-h5"
                                      >
                                        Profecient With
                                      </h5>
                                      <li
                                        class="MuiListItem-root MuiListItem-gutters"
                                      >
                                        <div
                                          class="MuiListItemIcon-root"
                                        >
                                          <svg
                                            aria-hidden="true"
                                            class="MuiSvgIcon-root"
                                            focusable="false"
                                            viewBox="0 0 24 24"
                                          >
                                            <circle
                                              cx="12"
                                              cy="12"
                                              r="8"
                                            />
                                          </svg>
                                        </div>
                                        <div
                                          class="MuiListItemText-root"
                                        >
                                          <span
                                            class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                                          >
                                            Class Components & Lifecycle Methods
                                          </span>
                                        </div>
                                      </li>
                                      <li
                                        class="MuiListItem-root MuiListItem-gutters"
                                      >
                                        <div
                                          class="MuiListItemIcon-root"
                                        >
                                          <svg
                                            aria-hidden="true"
                                            class="MuiSvgIcon-root"
                                            focusable="false"
                                            viewBox="0 0 24 24"
                                          >
                                            <circle
                                              cx="12"
                                              cy="12"
                                              r="8"
                                            />
                                          </svg>
                                        </div>
                                        <div
                                          class="MuiListItemText-root"
                                        >
                                          <span
                                            class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                                          >
                                            Functional Components with React Hooks
                                          </span>
                                        </div>
                                      </li>
                                      <li
                                        class="MuiListItem-root MuiListItem-gutters"
                                      >
                                        <div
                                          class="MuiListItemIcon-root"
                                        >
                                          <svg
                                            aria-hidden="true"
                                            class="MuiSvgIcon-root"
                                            focusable="false"
                                            viewBox="0 0 24 24"
                                          >
                                            <circle
                                              cx="12"
                                              cy="12"
                                              r="8"
                                            />
                                          </svg>
                                        </div>
                                        <div
                                          class="MuiListItemText-root"
                                        >
                                          <span
                                            class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                                          >
                                            State Management including React Context & Redux
                                          </span>
                                        </div>
                                      </li>
                                    </nav>
                                  </div>
                                </div>
                              </div>
                            </div>
                          </div>
                          <div
                            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
                          >
                            <div
                              class="MuiPaper-root MuiCard-root makeStyles-fullHeight-18 MuiPaper-elevation1 MuiPaper-rounded"
                            >
                              <div
                                align="center"
                                class="MuiGrid-root makeStyles-fullHeight-18 MuiGrid-container MuiGrid-direction-xs-column MuiGrid-justify-content-xs-center"
                              >
                                <div
                                  class="MuiGrid-root MuiGrid-item"
                                >
                                  <div
                                    class="MuiCardContent-root"
                                  >
                                    <div>
                                      <img
                                        class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img"
                                        src="/Images/Third-Party-Logos/Ruby_On_Rails_Logo.png"
                                        title="Zim Codes"
                                      />
                                    </div>
                                    <nav
                                      aria-label="main mailbox folders"
                                      class="MuiList-root MuiList-padding"
                                    >
                                      <h5
                                        class="MuiTypography-root MuiTypography-h5"
                                      >
                                        Building Reliable Backends With
                                      </h5>
                                      <li
                                        class="MuiListItem-root MuiListItem-gutters"
                                      >
                                        <div
                                          class="MuiListItemIcon-root"
                                        >
                                          <svg
                                            aria-hidden="true"
                                            class="MuiSvgIcon-root"
                                            focusable="false"
                                            viewBox="0 0 24 24"
                                          >
                                            <circle
                                              cx="12"
                                              cy="12"
                                              r="8"
                                            />
                                          </svg>
                                        </div>
                                        <div
                                          class="MuiListItemText-root"
                                        >
                                          <span
                                            class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                                          >
                                            ActiveRecord & SQL
                                          </span>
                                        </div>
                                      </li>
                                      <li
                                        class="MuiListItem-root MuiListItem-gutters"
                                      >
                                        <div
                                          class="MuiListItemIcon-root"
                                        >
                                          <svg
                                            aria-hidden="true"
                                            class="MuiSvgIcon-root"
                                            focusable="false"
                                            viewBox="0 0 24 24"
                                          >
                                            <circle
                                              cx="12"
                                              cy="12"
                                              r="8"
                                            />
                                          </svg>
                                        </div>
                                        <div
                                          class="MuiListItemText-root"
                                        >
                                          <span
                                            class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                                          >
                                            Building REST API Endpoints
                                          </span>
                                        </div>
                                      </li>
                                      <li
                                        class="MuiListItem-root MuiListItem-gutters"
                                      >
                                        <div
                                          class="MuiListItemIcon-root"
                                        >
                                          <svg
                                            aria-hidden="true"
                                            class="MuiSvgIcon-root"
                                            focusable="false"
                                            viewBox="0 0 24 24"
                                          >
                                            <circle
                                              cx="12"
                                              cy="12"
                                              r="8"
                                            />
                                          </svg>
                                        </div>
                                        <div
                                          class="MuiListItemText-root"
                                        >
                                          <span
                                            class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                                          >
                                            Connect to React Clients using Apollo & GraphQL
                                          </span>
                                        </div>
                                      </li>
                                    </nav>
                                  </div>
                                </div>
                              </div>
                            </div>
                          </div>
                          <div
                            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
                          >
                            <div
                              class="MuiPaper-root MuiCard-root makeStyles-fullHeight-18 MuiPaper-elevation1 MuiPaper-rounded"
                            >
                              <div
                                align="center"
                                class="MuiGrid-root makeStyles-fullHeight-18 MuiGrid-container MuiGrid-direction-xs-column MuiGrid-justify-content-xs-center"
                              >
                                <div
                                  class="MuiGrid-root MuiGrid-item"
                                >
                                  <div
                                    class="MuiCardContent-root"
                                  >
                                    <div>
                                      <img
                                        class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img"
                                        src="/Images/Third-Party-Logos/Gatsby-Logo.svg"
                                        title="Zim Codes"
                                      />
                                    </div>
                                    <nav
                                      aria-label="main mailbox folders"
                                      class="MuiList-root MuiList-padding"
                                    >
                                      <h5
                                        class="MuiTypography-root MuiTypography-h5"
                                      >
                                        Blazing Fast Sites Using
                                      </h5>
                                      <li
                                        class="MuiListItem-root MuiListItem-gutters"
                                      >
                                        <div
                                          class="MuiListItemIcon-root"
                                        >
                                          <svg
                                            aria-hidden="true"
                                            class="MuiSvgIcon-root"
                                            focusable="false"
                                            viewBox="0 0 24 24"
                                          >
                                            <circle
                                              cx="12"
                                              cy="12"
                                              r="8"
                                            />
                                          </svg>
                                        </div>
                                        <div
                                          class="MuiListItemText-root"
                                        >
                                          <span
                                            class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                                          >
                                            SSR Technology for performant static sites
                                          </span>
                                        </div>
                                      </li>
                                      <li
                                        class="MuiListItem-root MuiListItem-gutters"
                                      >
                                        <div
                                          class="MuiListItemIcon-root"
                                        >
                                          <svg
                                            aria-hidden="true"
                                            class="MuiSvgIcon-root"
                                            focusable="false"
                                            viewBox="0 0 24 24"
                                          >
                                            <circle
                                              cx="12"
                                              cy="12"
                                              r="8"
                                            />
                                          </svg>
                                        </div>
                                        <div
                                          class="MuiListItemText-root"
                                        >
                                          <span
                                            class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                                          >
                                            Custom Built Content Management using Netlify CMS
                                          </span>
                                        </div>
                                      </li>
                                      <li
                                        class="MuiListItem-root MuiListItem-gutters"
                                      >
                                        <div
                                          class="MuiListItemIcon-root"
                                        >
                                          <svg
                                            aria-hidden="true"
                                            class="MuiSvgIcon-root"
                                            focusable="false"
                                            viewBox="0 0 24 24"
                                          >
                                            <circle
                                              cx="12"
                                              cy="12"
                                              r="8"
                                            />
                                          </svg>
                                        </div>
                                        <div
                                          class="MuiListItemText-root"
                                        >
                                          <span
                                            class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                                          >
                                            Optimize Sites for High Ranking Lighthouse SEO Scores
                                          </span>
                                        </div>
                                      </li>
                                    </nav>
                                  </div>
                                </div>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="test-two"
                style="position: sticky; top: 0px; bottom: 0px; left: 0px; right: 0px; background-size: auto; background-repeat: no-repeat; will-change: transform; width: 100%; height: 0px; transform: translate3d(0,0px,0);"
              >
                <div
                  class="MuiGrid-root test-3 MuiGrid-item MuiGrid-grid-xs-12"
                >
                  <div
                    class="makeStyles-heroContainer-20"
                  >
                    <div
                      class="makeStyles-heroText-21"
                      id="hero-text"
                    >
                      Lets Build Together
                    </div>
                  </div>
                  <div
                    class="makeStyles-mainText-22"
                    id="scroll-down-text"
                  >
                    Scroll Down
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
